File: web/src/app/models/request/verifyPaymentRequest.ts
/**
 * What checkout hands back once Razorpay's widget reports success. `auth_verify_payment`
 * trusts none of it until `razorpaySignature` checks out server-side.
 */
export interface VerifyPaymentRequest {
  planId: string;
  razorpayOrderId: string;
  razorpayPaymentId: string;
  razorpaySignature: string;
}

File: src-tauri/src/models/request/verify_payment_request.rs
use serde::Deserialize;

/// What checkout hands back once Razorpay's widget reports success. The
/// `razorpay_signature` is what `verify-payment` recomputes and checks —
/// nothing here is trusted until that check passes.
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct VerifyPaymentRequest {
    pub plan_id: String,
    pub razorpay_order_id: String,
    pub razorpay_payment_id: String,
    pub razorpay_signature: String,
}
